crates/superposition_types/src/lib.rs

```diff
@@ -257,6 +257,46 @@ impl Display for RegexEnum {
     }
 }
 
+pub trait SuperpositionTenantConfig {
```

Comment:
This could be a struct with pub members right? Why define tenant config as a trait behaviour? No one can implement this elsewhere and pass it around. Why are we making it generic? Note: If the answer summarizes to "We will need this in the future", my response is "lets add it in the future". This is unecessary code complexity right now

Reply:
I'm guessing this is because of internal asks for auth? The tenant config struct internal to Juspay will be different I'm assuming